File: templates/python/fr/text/code/beeperLineBug1.py
# Fichier: Jeton LineBug.py
# ------------------------------
# Utilise un boucle while pour placer une ligne de jetons .
# Ce programme fonctionne pour un monde de toute taille.
# Cependant, parce que chaque monde en nécessite un de moins
# avancer s que put_ jetons il manque toujours un jeton .
from karel.stanfordkarel import *

# programme commence à main
def main():
   # se répète jusqu'à ce que Karel fasse face à un mur
   while l'avant_est_clair():
      # place un jeton sur la case actuelle
      laisser_jeton()
      # avancer au carré suivant
      avancer()

File: templates/java/fr/text/code/beeperLineBug1.java
/*
 * Fichier: JetonLigneJeton.java
 * ----------------------------
 * Utilise un boucle while pour placer une ligne de j
 * etons . Ce programme fonctionne pour un monde de t
 * oute taille. Cependant, comme chaque monde nécessi
 * te avancer moins que laisserJeton il manque toujou
 * rs un jeton .
 */
import stanford.karel.*;
public class JetonLigneJeton extends Karel {
   public void run() {
   	  /* répète jusqu'à ce que Karel soit face à un mur */
      while(devantDegage()) {
      	 /* placer un jeton sur le carré actuel */ 
         laisserJeton();
      	 /* avancer à la prochaine case */
         avancer();
      }
   }
}